PyPoE/poe/file/bundle.py

```diff
@@ -509,6 +528,22 @@ def _read(self, buffer: BytesIO):
                     directory_record.offset + directory_record.size
                 ]
             )
+        
+        self.hash_algorithm = None
+        if len(dirs := self.directories) > 0:
+            dir_iter = iter(dirs.values())
+            root_entry = next(dir_iter)
+            if root_entry.hash == 0x07e47507b4a92e53:
+                self.hash_algorithm = HASH_ALGORITHM.FNV1A64
```

Comment:
I quite like this bit

Reply:
This will run.. once? Like you're just getting the first item in the iterable for the entire run right?

Reply:
Yep, once per index opened which is once per run until we go and do something fun like add support for multiple releases in one process.